ui/src/app/machines/views/MachineDetails/MachineNetwork/BondForm/BondForm.tsx

```diff
@@ -80,40 +211,90 @@ const BondForm = ({
     : selected;
-  let editTooltip: string | null = null;
-  let editDisabled = false;
-  if (!editingMembers && validNics.length === 2) {
-    // Disable the button to add more members if there are no more to choose
-    // from.
-    editTooltip = "There are no additional valid members";
-    editDisabled = true;
-  } else if (editingMembers && selected.length < 2) {
-    // Don't let the user update the selection if they haven't chosen at least
-    // two interfaces.
-    editTooltip = "At least two interfaces must be selected";
-    editDisabled = true;
-  }
 
   return (
     <FormCard sidebar={false} stacked title="Create bond">
-      <InterfaceFormTable
-        editPrimary
-        interfaces={rows}
-        selected={selected}
-        selectedEditable={editingMembers}
-        setSelected={setSelected}
-        systemId={systemId}
-      />
-      <Tooltip message={editTooltip}>
-        <Button
-          data-test="edit-members"
-          disabled={editDisabled}
-          onClick={() => setEditingMembers(!editingMembers)}
-        >
-          {editingMembers ? "Update bond members" : "Edit bond members"}
-        </Button>
-      </Tooltip>
-      <button onClick={close}>Cancel</button>
+      <FormikForm
+        allowUnchanged
+        buttons={FormCardButtons}
+        cleanup={cleanup}
+        errors={errors}
+        initialValues={{
+          ...networkFieldsInitialValues,
+          bond_downdelay: 0,
+          bond_lacp_rate: "",
+          bond_mode: BondMode.ACTIVE_BACKUP,
+          bond_miimon: 0,
+          bond_updelay: 0,
+          bond_xmit_hash_policy: "",
+          fabric: vlan?.fabric || "",
+          linkMonitoring: "",
+          mac_address: firstNic?.mac_address || "",
+          name: nextName,
+          primary: firstNic?.id.toString(),
+          subnet: subnet?.id || "",
+          tags: [],
+          vlan: bondVLAN || "",
+        }}
+        onSaveAnalytics={{
+          action: "Create bond",
+          category: "Machine details networking",
+          label: "Create bond form",
+        }}
+        onCancel={close}
+        onSubmit={(values: BondFormValues) => {
+          // Clear the errors from the previous submission.
+          dispatch(cleanup());
+          type Payload = BondFormValues & {
+            parents: NetworkInterface["parents"];
+            system_id: Machine["system_id"];
+          };
+          const primary = toFormikNumber(values.primary);
+          const payload: Payload = {
+            ...values,
+            parents: primary ? generateParents(selected, primary) : [],
+            system_id: systemId,
+          };
+          const { linkMonitoring } = values;
+          Object.entries(payload).forEach(([key, value]) => {
+            if (
+              // Remove empty fields.
+              value === "" ||
+              // Remove fields that are not API values.
+              key === "primary" ||
+              key === "linkMonitoring" ||
```

Comment:
Interesting. I think I've typically only included submitted values in formik, and the rest just as component state.